src/equations/compressible_mhd_3d.jl

```diff
@@ -0,0 +1,185 @@
+    rho, v1, v2, v3, E, B1, B2, B3, psi = convert_transformed_to_primitive(u, equations)
+    # Here `gradients` is assumed to contain the gradients of the primitive variables (rho, v1, v2, v3, T)
+    # either computed directly or reverse engineered from the gradient of the entropy variables
+    # by way of the `convert_gradient_variables` function.
+
+    @unpack eta = equations
+
+    _, dv1dx, dv2dx, dv3dx, dTdx, dB1dx, dB2dx, dB3dx, _ = convert_derivative_to_primitive(u,
+                                                                                           gradients[1],
+                                                                                           equations)
+    _, dv1dy, dv2dy, dv3dy, dTdy, dB1dy, dB2dy, dB3dy, _ = convert_derivative_to_primitive(u,
+                                                                                           gradients[2],
+                                                                                           equations)
+    _, dv1dz, dv2dz, dv3dz, dTdz, dB1dz, dB2dz, dB3dz, _ = convert_derivative_to_primitive(u,
+                                                                                           gradients[3],
+                                                                                           equations)
+
+    # Components of viscous stress tensor
+
+    # Diagonal parts
+    # (4/3 * (v1)_x - 2/3 * ((v2)_y + (v3)_z)
+    tau_11 = 4.0 / 3.0 * dv1dx - 2.0 / 3.0 * (dv2dy + dv3dz)
+    # (4/3 * (v2)_y - 2/3 * ((v1)_x + (v3)_z)
+    tau_22 = 4.0 / 3.0 * dv2dy - 2.0 / 3.0 * (dv1dx + dv3dz)
+    # (4/3 * (v3)_z - 2/3 * ((v1)_x + (v2)_y)
+    tau_33 = 4.0 / 3.0 * dv3dz - 2.0 / 3.0 * (dv1dx + dv2dy)
+
+    # Off diagonal parts, exploit that stress tensor is symmetric
+    # ((v1)_y + (v2)_x)
+    tau_12 = dv1dy + dv2dx # = tau_21
+    # ((v1)_z + (v3)_x)
+    tau_13 = dv1dz + dv3dx # = tau_31
+    # ((v2)_z + (v3)_y)
+    tau_23 = dv2dz + dv3dy # = tau_32
+
+    # Fick's law q = -kappa * grad(T) = -kappa * grad(p / (R rho))
+    # with thermal diffusivity constant kappa = gamma μ R / ((gamma-1) Pr)
+    # Note, the gas constant cancels under this formulation, so it is not present
+    # in the implementation
+    q1 = equations.kappa * dTdx
+    q2 = equations.kappa * dTdy
+    q3 = equations.kappa * dTdz
+
+    # Constant dynamic viscosity is copied to a variable for readability.
+    # Offers flexibility for dynamic viscosity via Sutherland's law where it depends
+    # on temperature and reference values, Ts and Tref such that mu(T)
+    mu = equations.mu
+
+    if orientation == 1
+        # viscous flux components in the x-direction
+        f1 = zero(rho)
+        f2 = tau_11 * mu
+        f3 = tau_12 * mu
+        f4 = tau_13 * mu
+        f5 = (v1 * tau_11 + v2 * tau_12 + v3 * tau_13 + q1) * mu +
+             (-B3 * dB1dz + B3 * dB3dx + B2 * dB2dx - B2 * dB1dx) * eta
+        f6 = eta * dB1dx
+        f7 = eta * dB2dx
+        f8 = eta * dB3dx
```

Comment:
```suggestion
        f5 = (v1 * tau_11 + v2 * tau_12 + v3 * tau_13 + q1) * mu +
             (B2 * (dB2dx - dB1dy) + B3 * (dB3dx - dB1dz)) * eta
        f6 = zero(rho)
        f7 = eta * (dB2dx - dB1dy)
        f8 = eta * (dB3dx - dB1dz)
```
All of the new resistive terms are missing several components. I updated each Cartesian flux accordingly. This is how one would write them down on paper so they are likely computationally inefficient.